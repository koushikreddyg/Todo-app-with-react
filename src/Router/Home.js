import React from 'react';
import Form from '../components/Form';

class Home extends React.Component{
  constructor(props){
    super(props);
    this.state={
      options:[],
      error:false
    }
  }
Submit=(value)=>{
  const realvalue=value.trim();
  
  
}

render(){
  return(
    <div>
  
    <Form Submit={this.Submit}/>
    </div>
  )
}
}

export default Home;