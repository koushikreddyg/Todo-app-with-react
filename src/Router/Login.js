import React from 'react';
import {firebase,googleProvider,database} from '../firebase/firebase';
const Login=(props)=>(
    <div>
    <button onClick={(e)=>{
        firebase.auth().signInWithPopup(googleProvider).then(()=>{
            
        })
        
    }}>Login</button>
    </div>
)
export default Login