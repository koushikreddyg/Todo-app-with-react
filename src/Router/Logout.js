import React from 'react';
import {firebase,googleProvider,database} from '../firebase/firebase';
const Logout=(props)=>(
    <div>
    <button onClick={(e)=>{
        firebase.auth().signOut().then(()=>{
          
        })
       
    }}>Logout</button>
    </div>
)
export default Logout