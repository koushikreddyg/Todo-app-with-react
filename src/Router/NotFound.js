import React from 'react';
import {Link} from 'react-router-dom';
const NotFound=(props)=>(
  <div>
  This is not found page
  <Link to="/">Go Home</Link>
  </div>
)
export default NotFound;