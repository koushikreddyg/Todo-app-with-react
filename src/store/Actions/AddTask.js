import React from 'react';
import {database} from '../../firebase/firebase';
export  const addtask=(Task)=>({
    type:'ADD_TASK',
    Task:Task
});
export const addTaskFunction=(object)=>{
    return(dispatch,getState)=>{
        const uid=getState().Auth.uid;
        return database.ref(`users/${uid}/Tasks`).push({task:object}).then((ref)=>{
            dispatch(addtask({
                id:ref.key,
                task:object
            }));
        })
    }
    
}


