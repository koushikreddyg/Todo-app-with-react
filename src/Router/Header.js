import React from 'react';
import {NavLink} from 'react-router-dom';
import { Button } from 'antd';
import 'antd/dist/antd.css';
import Logout from './Logout';
 const Header=(props)=>(
  <div>
 <Button offset={6} type="primary" icon="cloud-download">
  <NavLink to="/dashboard" exact={true}>Home</NavLink>
 </Button>
 <Logout/>
  </div>
)
export default Header;