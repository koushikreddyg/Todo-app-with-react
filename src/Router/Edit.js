import React from 'react';
import Header from './Header';
import { Col } from 'antd';
import 'antd/dist/antd.css';
import { Input } from 'antd';
import { Button } from 'antd';
import Actions from '../Library/Actions';
import {editTaskFunction} from '../store/Actions/editTask';
import {removeTaskFunction} from '../store/Actions/RemoveTask';
import { connect } from 'react-redux'
const Search = Input.Search;


export class Edit extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      editValue: Actions.validateId(this.props.Tasks,this.props.ID)===true?
      this.props.editTask1.task:'' ,
      id: this.props.ID,
      error:'',
    }
  }
  editValueChange = (e) => {
    e.preventDefault();
    const editValue = e.target.value;
    this.setState(() => ({
      editValue
    }))
  }
  submitEditValue = (e) => {
    e.preventDefault();
    const editValue = this.state.editValue;
    const id = this.state.id;
    const Value=editValue.trim();
    let error=Actions.errorCheck(Value,this.props.Tasks)
    if(Value.length===0){
      this.setState(()=>({
        error: 'please enter some value'
      }))
    }else if(error.length>0){
      this.setState(()=>({
        error: 'this task is in your task list'
      }))
    }else{
      this.setState(()=>({
        error:undefined
      }))
      this.props.editTaskFunction(id, {task:Value} );
      this.setState(()=>({
        editValue:''
      }))
    this.props.history.push('/dashboard')
     }
    
  }
  Remove=(e)=>{
    e.preventDefault();
    this.props.removeTaskFunction(this.state.id);
    this.props.history.push('/dashboard');
  }
  componentDidMount(){
    if(this.state.editValue===''){
      this.props.history.push('/dashboard')
    }
  }

  render() {
    return (
      <div>
      <Header/>
        (<Col offset={7}>
          <form onSubmit={this.submitEditValue}>
            <Search value={this.state.editValue}
              style={{ width: 400 }}
              placeholder='please enter the to do task' name='addInput'
              onChange={this.editValueChange} />
            <Button type='primary' onClick={this.submitEditValue}>Edit!</Button><br />
            {this.state.error&&<p>{this.state.error}</p>}
          </form>
          <Button type='primary' span={20} onClick={this.Remove}>Remove Me!</Button>
        </Col>
      </div>
    );
  }
}
const mapDispatchToProps=(dispatch)=>({
  removeTaskFunction:(id)=>dispatch(removeTaskFunction(id)),
  editTaskFunction:(id,Value)=>dispatch(editTaskFunction(id,Value)),
})
const mapStateToProps = (state, props) => ({

  editTask1: state.Actions.find((item) => (item.id === props.match.params.id)),
  ID:props.match.params.id,
  Tasks:state.Actions,
})
export default connect(mapStateToProps,mapDispatchToProps)(Edit);
