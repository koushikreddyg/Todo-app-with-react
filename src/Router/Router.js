import React from 'react';
import Header from './Header';
import Edit from './Edit';
import NotFound from './NotFound';
import Home from './Home';
import Login from './Login'
import Logout from './Logout';
import {Router, Switch, Route} from 'react-router-dom';
import createHistory from 'history/createBrowserHistory';
export const history=createHistory()
 const AppRouter=()=>(
  <div>
  <Router history={history}>
  <div>
        <Switch>
          <Route path="/" component={Login} exact={true}/>
          <Route path="/dashboard" component={Home} />
          <Route path="/edit/:id" component={Edit}/>
          <Route  component={NotFound}/>
        </Switch>
        </div>
      </Router>
      
  
  </div>
)
export default AppRouter;