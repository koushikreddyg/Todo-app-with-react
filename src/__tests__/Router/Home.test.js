import React from 'react';
import {shallow} from 'enzyme';
import {Home} from '../../Router/Home';
import Tasks from '../../fixtures/Tasks';
import configureMockStore from 'redux-mock-store';
import Thunk from 'redux-thunk';
import {addTaskFunction} from '../../store/Actions/AddTask';

test('Home page with change of input',()=>{
    const wrapper=shallow(<Home />)
    const value='koushik';
    expect(wrapper).toMatchSnapshot();
    wrapper.find('Search').simulate('change',{
        target:{value},
        preventDefault:()=>{}
    })
    expect(wrapper.state('task')).toBe(value)
})
test('Home page with Submit button simulated',()=>{
    const addTaskFunction=jest.fn();
    const task='FIFA';
    const wrapper=shallow(<Home addTaskFunction={addTaskFunction} Tasks={Tasks}/>);
    expect(wrapper).toMatchSnapshot();
    wrapper.find('Search').simulate('change',{
        target:{value:task},
        preventDefault:()=>{}
    })
    wrapper.find('form').simulate('submit',
   { preventDefault:()=>{}}
)
expect(wrapper.state('error')).toBe(undefined)
expect(addTaskFunction).toHaveBeenLastCalledWith("FIFA");

})
// test('test with mock store in dispatch add Task',()=>{
//     const mockStore=configureMockStore({});
//     const store=mockStore({});
//     store.dispatch(AddTask(Tasks[0]));
//     const actions=store.getActions();
//     expect(actions[0]).toEqual({
//         type:'ADD_TASK',
//         Task:Tasks[0]
//     });
// })