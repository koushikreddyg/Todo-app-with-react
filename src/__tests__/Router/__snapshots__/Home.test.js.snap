// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page with Submit button simulated 1`] = `
<div>
  <Header />
  <Row
    gutter={0}
  >
    <form
      onSubmit={[Function]}
    >
      <Col
        offset={7}
      >
        <Search
          enterButton={false}
          inputPrefixCls="ant-input"
          name="addInput"
          onChange={[Function]}
          placeholder="please enter the to do task"
          prefixCls="ant-input-search"
          style={
            Object {
              "width": 325,
            }
          }
          value=""
        />
        <Button
          ghost={false}
          loading={false}
          onClick={[Function]}
          prefixCls="ant-btn"
          type="primary"
        >
          Submit
        </Button>
      </Col>
    </form>
  </Row>
  <Row
    gutter={0}
  >
    <Col
      offset={7}
    >
      <div
        style={
          Object {
            "background": "#ECECEC",
            "padding": "30px",
            "textAlign": "left",
            "width": 400,
          }
        }
      >
        <Card
          bordered={true}
          style={
            Object {
              "width": 330,
            }
          }
          title="Task List"
        >
          <DisplayTasks
            Tasks={
              Array [
                Object {
                  "id": 24,
                  "task": "Cricket",
                },
                Object {
                  "id": 25,
                  "task": "Football",
                },
                Object {
                  "id": 26,
                  "task": "Hockey",
                },
              ]
            }
            bordered={true}
          />
        </Card>
      </div>
    </Col>
  </Row>
</div>
`;

exports[`Home page with change of input 1`] = `
<div>
  <Header />
  <Row
    gutter={0}
  >
    <form
      onSubmit={[Function]}
    >
      <Col
        offset={7}
      >
        <Search
          enterButton={false}
          inputPrefixCls="ant-input"
          name="addInput"
          onChange={[Function]}
          placeholder="please enter the to do task"
          prefixCls="ant-input-search"
          style={
            Object {
              "width": 325,
            }
          }
          value=""
        />
        <Button
          ghost={false}
          loading={false}
          onClick={[Function]}
          prefixCls="ant-btn"
          type="primary"
        >
          Submit
        </Button>
      </Col>
    </form>
  </Row>
  <Row
    gutter={0}
  >
    <Col
      offset={7}
    >
      <div
        style={
          Object {
            "background": "#ECECEC",
            "padding": "30px",
            "textAlign": "left",
            "width": 400,
          }
        }
      >
        <Card
          bordered={true}
          style={
            Object {
              "width": 330,
            }
          }
          title="Task List"
        >
          <DisplayTasks
            bordered={true}
          />
        </Card>
      </div>
    </Col>
  </Row>
</div>
`;
